src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -2567,6 +2596,25 @@ public virtual void UpdateObject(object entity)
             this.UpdateObjectInternal(entity, false /*failIfNotUnchanged*/);
         }
 
+        /// <summary>
+        /// Convert array object to List of strings.
+        /// </summary>
+        /// <param name="dependsOnIds">Object array of DependsOnIds.</param>
+        /// <returns>List of strings.</returns>
+        private static List<string> ConvertDependOnIdsToString(object[] dependsOnIds)
+        {
+            var dependsOn = new List<string>();
+            if (dependsOnIds != null)
+            {
+                foreach (var id in dependsOnIds)
+                {
+                    dependsOn.Add(id.ToString());

```

Comment:
```suggestion
                    yield return id.ToString();

```

Reply:
See this comment https://github.com/OData/odata.net/pull/1806#discussion_r610376573